gapic-generator-java/src/main/java/com/google/api/generator/gapic/model/Service.java

```diff
@@ -184,6 +190,17 @@ private static String parseApiVersion(String protoPackage) {
     return apiVersion;
   }
 
+  // Parse the service name from the default host configured in the protos
+  // or service yaml file. For Google Cloud Services, the default host value
+  // is expected to contain `.googleapis.com`. Exceptions may exist (i.e. localhost),
+  // in which case we will return an empty string.
+  private static String parseHostServiceName(String defaultHost) {
```

Comment:
I know we are already testing this method indirectly through golden tests. Since this is in Service not composers, and we can easily mock the input, it would be great if we can have unit tests testing it directly as well.

Reply:
Added some tests for parsing, thanks!